### PR TITLE
Add AddLabel()/AddAnnotation() for Route

### PR DESCRIPTION
**Pull request by elvgarrui**
I think this might need testing, but looking at similar PR (https://github.com/openstack-k8s-operators/lib-common/pull/418) I did not fully understand how I should create something similar to what we see there in "modules/common/service/service_test.go" for the Route struct.

### CODE REVIEW

**Review by stuggi — COMMENTED** [review]
right, this needs tests as in https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/service/service_test.go#L479 . specify a default Route starting with no annotation, then:
- add a annotation
- add another annotation should have 2
- override an annotation (first one remains)

you could just make a copy of service.TestOverrideSpecAddAnnotation and change it for the Route use case.

File: modules/common/route/route.go [hunk 1]
```diff
@@ -189,3 +189,20 @@ func (r *Route) Delete(
 
 	return nil
 }
+
+// AddAnnotation - Adds annotation and merges it with the current set
+func (r *Route) addAnnotation(anno map[string]string) {
```
[the remark this answers is not in the document]
Comment (elvgarrui) [reply modules/common/route/route.go]: Thanks for pointing it out, fixed

File: modules/common/route/route.go [hunk 2]
```diff
@@ -189,3 +189,20 @@ func (r *Route) Delete(
...
+       if r.EmbeddedLabelsAnnotations == nil {
+               r.EmbeddedLabelsAnnotations = &EmbededLabelsAnnotations{}
+       }
+       r.Annotations = util.MergeMaps(r.Annotations, anno)
+}
+
+
+// AddLabel - Adds annotation and merges it with the current set
+func (r *Route) addLabel(label map[string]string) {
```
[the remark this answers is not in the document]
Comment (elvgarrui) [reply modules/common/route/route.go]: Done!

File: modules/common/route/route.go [hunk 3]
```diff
@@ -189,3 +189,24 @@ func (r *Route) Delete(
 
 	return nil
 }
+
+// AddAnnotation - Adds annotation and merges it with the current set
+func (r *Route) AddAnnotation(anno map[string]string) {
+	r.route.Annotations = util.MergeStringMaps(r.route.Annotations, anno)
+}
+
```
Comment (stuggi) [inline modules/common/route/route.go:197]: I guess we also want an `AddLabel` for `Route`?
```
// AddLabel - Adds label and merges it with the current set
func (r *Route) AddLabel(label map[string]string) {
	r.route.Labels = util.MergeStringMaps(r.route.Labels, label)
}
```

File: modules/common/route/route.go [hunk 4]
```diff
@@ -189,3 +189,24 @@ func (r *Route) Delete(
...
+// AddAnnotation - Adds annotation and merges it with the current set
+func (r *OverrideSpec) AddAnnotation(anno map[string]string) {
+	if r.EmbeddedLabelsAnnotations == nil {
+		r.EmbeddedLabelsAnnotations = &EmbeddedLabelsAnnotations{}
+	}
+	r.Annotations = util.MergeMaps(r.Annotations, anno)
+}
+
+// AddLabel - Adds annotation and merges it with the current set
```
Comment (stuggi) [inline modules/common/route/route.go:206]: nit, `Adds annotation` -> `Adds label`

File: modules/common/route/route.go (hunk 3 above)
Comment (elvgarrui) [reply modules/common/route/route.go:207]: Done!

File: modules/common/route/route.go (hunk 4 above)
Comment (elvgarrui) [reply modules/common/route/route.go:221]: thx!

**Review by stuggi — APPROVED** [review]
/lgtm thanks!

**Pull request merged**